realm/realm-annotations-processor/src/main/java/io/realm/processor/Backlink.java

```diff
@@ -16,34 +16,43 @@
 
 package io.realm.processor;
 
+import javax.lang.model.element.Modifier;
 import javax.lang.model.element.VariableElement;
 
 import io.realm.annotations.LinkingObjects;
 import io.realm.annotations.Required;
 
 /**
- * A Backlink is:
+ * A <b>Backlink</b> is an implicit backwards reference.  If field <code>sourceField</code> in instance <code>I</code>
+ * of type <code>SourceClass</code> holds a reference to instance <code>J</code> of type <code>TargetClass</code>,
+ * then a "backlink" is the automatically created reference from <code>J</code> to <code>I</code>.
+ * Backlinks are automatically created and destroyed when the forward references to which they correspond are
+ * created and destroyed.  This can dramatically reduce the complexity of client code.
  * <p>
+ * To expose backinks for use, create a declaration as follows:
  * <code>
  * class TargetClass {
- * // ...
- * {@literal @}LinkingObjects("sourceField")
- * RealmResults&lt;SourceClass&gt; targetField;
+ *     // ...
+ *     {@literal @}LinkingObjects("sourceField")
+ *     final RealmResults&lt;SourceClass&gt; targetField = null;
  * }
  * </code>.
  * <p>
- * The targetField of a managed object cannot be assigned.  It can be queried normally.
- * When an instance X of a class containing a field annotated with the <code>@LinkingObjects</code> annotation
- * is managed (<code>copyToRealm()</code>), the newly managed object will contain, in the annotated field,
- * references to any instances of <code>sourceClass</code> whose <code>sourceField</code> contains a reference to X.
- * Any previous contents of the targetField are lost.
+ * The targetField, the field annotated with the {@literal @}LinkingObjects annotation must be final.
```

Comment:
This requirement is more important in the `@LinkingObjects` Javadoc since that is the public API docs. .